compiler/src/iree/compiler/ExternalInterfaces/UtilExternalModels.cpp

```diff
@@ -371,13 +371,10 @@ struct HoistableLinalgOpInterface
       return !isa<linalg::FillOp>(op);
     }
 
-    // Don't hoist ops with no inputs, their result is defined by `linalg.index`
-    // ops and should be fused with their consumers.
-    if (genericOp.getNumDpsInputs() == 0) {
-      return false;
-    }
-
-    if (linalg::isaFillOpInterface(genericOp).has_value()) {
+    // Don't hoist ops with no tensor inputs, they can be fused with their
```

Comment:
The comment seems to be carrying more information than what the code is checking for.